upf/upf_pfcp_api.c

```diff
@@ -857,11 +1075,41 @@ handle_create_pdr (upf_session_t * sx, pfcp_create_pdr_t * create_pdr,
 	create->pdi.nwi_index = nwi - gtm->nwis;
       }
 
+    p = hash_get_mem (gtm->nwi_index_by_name, nwi->name);
+    if (!p)
+      {
+	upf_debug("Can't find nwi idx by name\n");
+        r = -1;
+	break;
+      }
+
+    idx = p[0];
+    res = &gtm->upip_res[idx];
```

Comment:
wrong, you need to lookup `res` on `upip_res_index` using the nwi index (which be calculates as `nwi - gtp-nwis`).

Although, I have to admit that storing the UPIP information in a pool is a bit over complicating things. A simple vector that uses the same index as the NWI would be much simpler. Alternatively, storing the UPIP information directly in the NWI struct would also work nicely.
Anyhow, I you want to change it, do it in a separate commit before the FTUP changes.